src/client.js

```diff
@@ -217,11 +217,10 @@ class Client extends events.EventEmitter {
   }
 
   /**
-   * Update the _recycleTimer, either to expire at recycleAt or, if that's omitted,
-   * after recycleInterval.
+   * Update the _recycleTimer based on recycleInterval
    */
-  async _updateRecycleTimer(recycleAt) {
-    const recycleAfter = recycleAt ?  recycleAt - new Date() : this._recycleInterval;
+  async _updateRecycleTimer() {
```

Comment:
This function was neccessary to deal with recycleAfter from tc-pulse. So I think this function now unnecessarily complicates logic unless there is some compatibility issue.

Reply:
We need some logic to handle this timer, and this seems like a good place for it.  What alternative would you suggest?

Reply:
Use `setInterval(this.recycle, this._recycleInterval)` in the `Client` constructor as before.